src/web/app/components/session-edit-form/session-edit-form.component.spec.ts

```diff
@@ -53,4 +57,117 @@ describe('SessionEditFormComponent', () => {
+    component.model.submissionStartTime = time;
+    component.triggerSubmissionOpeningDateModelChange(submissionStartDateField, date);
+    expect(triggerModelChangeSpy).toHaveBeenCalledWith(submissionStartDateField, date);
+  });
+
+  it('should emit a model change event with the updated field when triggerModelChange is called', () => {
+    const field = 'courseId';
+    const data = 'testId';
+    const modelChangeSpy = jest.spyOn(component.modelChange, 'emit');
+    component.triggerModelChange(field, data);
+    expect(modelChangeSpy).toHaveBeenCalledWith({
+      ...component.model,
+      [field]: data,
+    });
+  });
+
+  it('should emit modelChange event when a valid course ID is provided', () => {
+    const newCourseId = 'testId1';
+    const courseCandidates: Course[] = [
+      {
+        courseId: 'testId1',
+        courseName: 'testCourse1',
+        timeZone: 'Asia/Singapore',
+        institute: 'Institute 1',
+        creationTimestamp: 1000000000000,
+        deletionTimestamp: 1500000000000,
+      },
+    ];
+    component.courseCandidates = courseCandidates;
+    const modelChangeSpy = jest.spyOn(component.modelChange, 'emit');
+    component.courseIdChangeHandler(newCourseId);
+    expect(modelChangeSpy).toHaveBeenCalledWith({
+      ...component.model,
+      courseId: newCourseId,
+      courseName: 'testCourse1',
+      timeZone: 'Asia/Singapore',
+    });
+  });
+
+  it('should not emit modelChange event when no candidates are found', () => {
+    const newCourseId = 'testId1';
+    const courseCandidates: Course[] = [];
+    component.courseCandidates = courseCandidates;
+    const modelChangeSpy = jest.spyOn(component.modelChange, 'emit');
+    component.courseIdChangeHandler(newCourseId);
+    expect(modelChangeSpy).not.toHaveBeenCalled();
+  });
+
+  it('should emit addNewSessionEvent when session edit form mode is ADD', () => {
+    component.formMode = SessionEditFormMode.ADD;
+    const addNewSessionSpy = jest.spyOn(component.addNewSessionEvent, 'emit');
+    component.submitFormHandler();
+    expect(addNewSessionSpy).toHaveBeenCalled();
+  });
+
+  it('should emit a cancelEditingSession event after modal confimation', async () => {
+    jest.spyOn((component as any).simpleModalService, 'openConfirmationModal').mockReturnValue(mockModal);
+    const cancelEditingSessionSpy = jest.spyOn(component.cancelEditingSessionEvent, 'emit');
+    component.cancelHandler();
+    await mockModal.result;
+    expect(cancelEditingSessionSpy).toHaveBeenCalled();
+  });
+
+  it('should emit a deleteExistingSession event after modal confimation', async () => {
+    jest.spyOn((component as any).simpleModalService, 'openConfirmationModal').mockReturnValue(mockModal);
+    const deleteExistingSessionSpy = jest.spyOn(component.deleteExistingSessionEvent, 'emit');
+    component.deleteHandler();
+    await mockModal.result;
+    expect(deleteExistingSessionSpy).toHaveBeenCalled();
+  });
```

Comment:
for these methods, we should be creating a modalSpy and checking if it's called with the appropriate params, you can see admin-notifications-page.component.spec.ts for an example